app/src/main/java/org/p2p/wallet/send/model/SendSolanaFee.kt

```diff
@@ -143,10 +151,13 @@ data class SendSolanaFee constructor(
                 if (desiredAmount != null) ReduceInputAmount(desiredAmount) else SwitchToSol
             }
             // if there is enough SPL token balance to cover amount and fee
-            !isSourceSol && sourceTokenTotal.isMoreThan(totalNeeded) ->
-                UpdateFeePayer
-            else ->
-                SwitchToSol
+            !isSourceSol && sourceTokenTotal.isMoreThan(totalNeeded) -> UpdateFeePayer
```

Comment:
You should delete `UpdateFeePayer` type and change it to `SwitchToSpl(sourceToken)`
`UpdateFeePayer` is redundant and useless